src/Model/Host.php

```diff
@@ -21,59 +21,91 @@
 
 namespace Friendica\Model;
 
-use Friendica\Core\Logger;
-use Friendica\Database\DBA;
+use Friendica\Database\Database;
+use Psr\Log\LoggerInterface;
 
 class Host
 {
+	/** @var string db table of host model */
+	const TABLE = 'host';
+
 	/**
-	 * Get the id for a given hostname
-	 * When empty, the current hostname is used
+	 * Defines the environment variable, which includes the current node name instead of the detected hostname
 	 *
-	 * @param string $hostname
+	 * @var string
 	 *
-	 * @return integer host name id
-	 * @throws \Exception
+	 * @notice This is used for cluster environments, where the each node defines it own hostname.
 	 */
-	public static function getId(string $hostname = '')
+	const ENV_VARIABLE = 'NODE_NAME';
+
+	/** @var integer The host id */
+	private $id = -1;
+	/** @var string The host name */
+	private $name;
+
+	/** @var LoggerInterface */
+	private $logger;
+
+	public function __construct(Database $dba,  LoggerInterface $logger, array $server = [])
+	{
+		$this->logger = $logger;
+
+		$this->detectHostname($dba, $server);
+	}
+
+	private function detectHostname(Database $dba, array $server)
 	{
+		$hostname = $server[self::ENV_VARIABLE] ?? null;
```

Comment:
When I understood it correctly, then this `server` variable is only used during the tests? If yes, then why is it an array and not simply a string?

Reply:
No, in production the `$server` parameter of the constructor is provided by Dice according to the rules in `static/dependencies.config.php`.

Reply:
Then I recommend reducing this to always use the appropriate PHP function and no environment variable. It adds some uncertainy. The CLI PHP could fetch its hostname via another way than the frontend PHP process. And since they then origin from a different source, we cannot guarantee for sure that the values on the same machine were identical.

Reply:
What is the appropriate PHP function? In a containerized environment (like it is the target), the environment variable would be set automatically by the Docker script.

Reply:
I meant the PHP function that is used when the variable isn't set.

Reply:
The null-coalescing operator `??` is an appropriate PHP function when a variable isn't set. What did you have in mind?

Reply:
No. I meant to use the PHP function to fetch the hostname all the time and not only when it isn't found in the `$_SERVER` array.

Reply:
I don't think it's necessary since host names wouldn't be used outside of a containerized setup but maybe @nupplaphil can shed more light on this particular code decision.

Reply:
Well, like I said: The mixture of having a frontend PHP process and a backend PHP process could lead to a situation where the hostnames are detected differently. And this could cause weird problems. I think that there is really no reason to not always use the PHP function to fetch the hostname.

Reply:
Not with an environment variable that was set by Docker. It still would be the same (virtual) machine (and its environment variables) whether it's a frontend process or a backend process.

Reply:
But this does have to run on regular systems as well.

Reply:
On single-host nodes, the hosts table wouldn't matter so it's fine if this code returns null because the environment variable wouldn't be set.

Reply:
This code wouldn't return `null` because it falls back on the PHP function for fetching the hostname.

Reply:
`php_uname` is **the** way to retrieve the hostname in a "normal" PHP environment. But for cluster environments, we cannot trust the `hostname` entry of a server anymore. I found references where the cluster-manager sets `/etc/hostname` to `localhost`. So the usage of `php_uname` is not adviced from cluster experts.

Instead they're using this method:
https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/#use-pod-fields-as-values-for-environment-variables , in particular this section:
```yml
      - name: MY_POD_NAME
          valueFrom:
            fieldRef:
              fieldPath: metadata.name
```
which is setting the `metadata.name` of a node (= the unique name of the node) as the envionment variable `MY_POD_NAME`. So I do need a way to disable `php_uname` and instead use a environment variable for it.

I thought doing it inside the `Host` constructor is a good idea. But if we really want a own "repository" class (see https://github.com/friendica/friendica/pull/9114#discussion_r484103828 ) I'm with you that the decision if using `php_uname` or the environment variable, should be done outside of the Host class itself.

Reply:
I would suggest some configuration that defaults to `gethostname()` and only if configured, it should take the value from a different source. I guess that people who can set up these clusters can handle this.

Reply:
There's now a "selectCurrentHost()" method in the repository, which either uses `gethostname()` (as suggested instead of the `php_uname('n')` method or checks the env-variable